app/controllers/serviceLearning/routes.py

```diff
@@ -112,28 +116,36 @@ def withdrawCourse(courseID):
         else:
             flash("Unauthorized to perform this action", 'warning')
     except Exception as e:
+        print(e)
         flash("Withdrawal Unsuccessful", 'warning')
     return ""
 
+@serviceLearning_bp.route('/serviceLearning/renew/<courseID>/<termID>/', methods = ['POST'])
+def renewCourse(courseID, termID):
+    try:
+        if g.current_user.isAdmin or g.current_user.isFaculty:
+            renewProposal(courseID, termID)
+            flash("Course successfully renewed", 'success')
+        else:
+            flash("Unauthorized to perform this action", 'warning')
+    except Exception as e:
+        print(e)
+        flash("Renewal Unsuccessful", 'warning')
+    return ""
+
 @serviceLearning_bp.route('/serviceLearning/approveCourse/', methods=['POST'])
 def approveCourse():
     """
     This function updates and approves a Service-Learning Course when using  the
         approve button.
     return: empty string because AJAX needs to receive something
-    """ 
-    if len(request.form)==1:
-        course=Course.get_by_id(request.form['courseID']) # if only course is reviewed pass the course ID
-        
-    elif 'courseID' in request.form:
-        course = updateCourse(request.form.copy(), instructorsDict) # if edit course, Updates database with the completed fields and get course ID
-        
-    else:
-        course=createCourse(request.form.copy(), instructorsDict) # creat course first and get its ID to approve next
+    """
     try:
-        course.status = 2
-        course.save() # saves the query and approves course in the database
-        flash("Course approved!", "success")        
+        updateCourse(request.form.copy(), instructorsDict) # Updates database with the completed fields
+        # The next line creates the query to approve the course
+        course = Course.update(status = 2).where(Course.id == request.form['courseID'])
+        course.execute() # Executes the query and approves course in the database
+        flash("Course approved!", "success")
     except:
-        flash("Course not approved!", "danger")  
+        flash("Course not approved", "warning")
```

Comment:
This function seems like a reversion of what Ala was working on. It no longer handles the 3 different cases properly. Are those cases still relevant?

Reply:
resolved